### DISCUSSION TITLE
Problems with fail2ban not working

### DISCUSSION THREAD

**Discussion by wyfang** [dani-garcia/vaultwarden#4489] [Q&A]
I'm running it in docker, here's my docker run command:
`docker run -d --name vaultwarden \
-v /home/Wi-Fi/Bitwarden/:/data/  \
-v /home/Wi-Fi/Bitwarden/log/:/data/log/ \
-v /etc/localtime:/etc/localtime:ro \
-p 127.0.0.1:8001:80 \
--restart=always \
-e LOG_FILE=/data/log/vaultwarden.log \
vaultwarden/server:latest`

I can now see in the host under /home/Wi-Fi/Bitwarden/log/vaultwarden.log
`[2024-04-10 19:27:19.785][vaultwarden::api::identity][ERROR] Username or password is incorrect. Please retry. ip: 172.22.159.50. username: xxxxxxx@gmail.com.`

Then I followed the documentation and created a new
/etc/fail2ban/filter.d/vaultwarden.log
```
[INCLUDES] (contains
before = common.conf

Definitions
failregex = ^. *Username or password is incorrect\. Try again\. IP: <ADDR>\. Username:. *Username:.
ignoreregex =
```
and

/etc/fail2ban/jail.d/vaultwarden.local
```
[vaultwarden]
enabled = true
port = 80,443,8081
filter = vaultwarden
chain = FORWARD
banaction = %(banaction_allports)s
logpath = /home/Wi-Fi/Bitwarden/log/vaultwarden.log
maxretry = 3
bantime = 14400
findtime = 14400
```

Then I reloaded fail2ban and then I tried several times to enter the wrong account password and saw the corresponding logs, but it was not blocked,I can still access the page normally even though there is an ip being banned.
```

`[root@VM-4-16-opencloudos ~]# sudo fail2ban-client status vaultwarden
Status for the jail: vaultwarden
|- Filter
|  |- Currently failed: 3
|  |- Total failed:     18
|  `- File list:        /home/Wi-Fi/Bitwarden/log/vaultwarden.log
`- Actions
   |- Currently banned: 2
   |- Total banned:     2
   `- Banned IP list:   172.22.159.50 172.22.158.25`

```
------------------------

By the way, I'd like to ask about this.
/home/Wi-Fi/Bitwarden/log/vaultwarden.log file will be infinitely large? Is there any way I can limit its maximum size or keep it for only 1 day?

**Comment by stefan0xC** [comment]
From the IP range (which was [reserved for private use](https://www.rfc-editor.org/rfc/rfc1918.html#section-3) and is probably from your docker networks) my guess is that you are not propagating the real IP address to Vaultwarden. So you might want to look into that.

Regarding the second question: You would be limited by the available disk space. You can set up logrotate. Cf. [this example in the wiki](https://github.com/dani-garcia/vaultwarden/wiki/Logrotate-example). Alternatively (instead of setting `LOG_FILE` where it is assumed that you will take care of the problem) you can look into setting up fail2ban with systemd or using [dockers local file logging driver](https://docs.docker.com/config/containers/logging/local/).

**Reply by wyfang** [reply]
WOW, thanks so much for the answer! It's my fault, I ignored the setting CF-Connecting-IP in the wiki. also it has been possible to ban to the correct ip.
